ibis/pandas/core.py

```diff
@@ -237,24 +286,27 @@ def execute_until_in_scope(
     if op in new_scope:
         return new_scope
 
-    # figure out what arguments we're able to compute on based on the
-    # expressions inputs. things like expressions, None, and scalar types are
-    # computable whereas ``list``s are not
-    computable_args = [arg for arg in op.inputs if is_computable_input(arg)]
-
     # recursively compute each node's arguments until we've changed type
+
+    if len(arg_timecontexts) != len(computable_args):
```

Comment:
where is this hit?

Reply:
So compute_time_context should return with a list with the same as computable_args, the two lists will be zipping together for further execution. This is referenced in line#311 ```for (arg, timecontext) in zip(computable_args, arg_timecontexts)```
I also added a comment explaining this.